### ISSUE TITLE
false positive on reverse() of custom class implementing Sequence protocol

### ISSUE THREAD

**Issue by mrwonko** [python/typeshed#6098]
**Bug Report**

I have a custom class that implements `__len__` and `__getitem__` as described in [the documentation of `reversed`](https://docs.python.org/3.8/library/functions.html#reversed), thereby fulfilling the `Sequence` protocol. mypy complains that it is not a valid argument to `reversed`.

**To Reproduce**

```Python
class CustomReversible:
    def __init__(self) -> None:
        self.__data = ["foo", "bar"]

    def __len__(self) -> int:
        return len(self.__data)

    def __getitem__(self, item) -> str:
        return self.__data[item]


print(list(reversed(CustomReversible())))
```

**Expected Behavior**

```txt
Success: no issues found in 1 source file
```

**Actual Behavior**

```txt
mypy_repro.py:12: error: No overload variant of "reversed" matches argument type "CustomReversible"  [call-overload]
    print(list(reversed(CustomReversible())))
               ^
mypy_repro.py:12: note: Possible overload variants:
mypy_repro.py:12: note:     def [_T] reversed(self, Sequence[_T]) -> reversed[_T]
mypy_repro.py:12: note:     def [_T] reversed(self, Reversible[_T]) -> reversed[_T]
Found 1 error in 1 file (checked 1 source file)
```

Explicitly inheriting from `Sequence[str]` helps, but it is my understanding that you are not supposed to have to inherit from protocols to implement them.

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.8.2
- Operating system and version: macOS Big Sur 11.6

**Comment by srittau** [comment]
`_typeshed.SupportsLenAndGetItem` seems to fit. In fact, https://docs.python.org/3.8/glossary.html#term-sequence says:

> sequence
> An iterable which supports efficient element access using integer indices via the `__getitem__()` special method and defines a `__len__()` method that returns the length of the sequence. [...]
> The collections.abc.Sequence abstract base class defines a much richer interface that goes beyond just `__getitem__()` and `__len__()`, adding `count()`, `index()`, `__contains__()`, and `__reversed__()`. [...]

This makes me think that we should rename `SupportsLenAndGetItem` to something like `SimpleSequence` and explicitly link to the docs.

**Comment by Akuli** [comment]
This was fixed a couple months ago in #5820, but I think mypy is using a really old typeshed.

**Comment by JelleZijlstra** [comment]
Good catch, and I merged that PR too :) I must be getting old.

In that case, this should get fixed by the release of mypy 0.920, which is expected soon.
